tests/stub_terminal.h

```diff
@@ -61,10 +63,26 @@ class StubTerminal : public multipass::Terminal
     {
     }
 
+    ConsolePtr set_console(ConsolePtr new_console)
+    {
+        return std::exchange(console, std::move(new_console));
+    }
+
+    ConsolePtr make_console(ssh_channel channel) override
+    {
+        if (!console)
+        {
+            return std::make_unique<StubConsole>();
+        }
+
+        return std::exchange(console, nullptr);
+    }
```

Comment:
Hmm, this smells a little of over-engineering :nose: Is `set_console` even used anywhere? If it ever was, wouldn't a mock be more appropriate than a stub? Why don't you simplify to just this?

```suggestion
    ConsolePtr make_console(ssh_channel channel) override
    {
        return std::make_unique<StubConsole>();
    }
```

Reply:
Good point